src/gui/session.py

```diff
@@ -22,5 +24,27 @@ def unload(self):
         del self.action
 
     def run(self):
-        QMessageBox.information(None, "Minimal plugin",
-                                "Do something Fred here")
+        workingDirectory = QFileDialog.getExistingDirectory(
+            None, "Select a working directory", QDir.currentPath()
+        )
+        if not workingDirectory:
+            # no directory selected
+            return
+
+        if not self.session.isDataBaseAvailable(workingDirectory):
+            # No geopackage DB in this directory
+            self.session.createEmptyDataBase(workingDirectory)
+
+        self.session.directoryPath = workingDirectory
+
+    def addLayerFromDataBase(self):
```

Comment:
unused method?

Reply:
Not really. It is useless for now (because we don't add the layer anymore) but it will be in few PR so I propose to keep it